hilti/toolchain/include/compiler/plugin.h

```diff
@@ -304,4 +304,6 @@ class Register {
 
 } // namespace plugin
 
+Plugin hilti_plugin();
```

Comment:
Let's move this into `detail` and rename to something like `create_hilti_plugin` to avoid somebody calling this to get access to an already existing plugin instance. Same below for Spicy.